react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { clicks: 0 };
+    this.handleClicks = this.handleClicks.bind(this);
+    this.renderButton = this.renderButton.bind(this);
+  }
+
+  handleClicks() {
+    this.setState(({ clicks }) => ({
+      clicks: clicks + 1
+    }));
+    const button = document.querySelector('.hot-button');
```

Comment:
We almost never want to use the `document` object in react. We don't want to make direct changes to the DOM without React knowing. Also you can't check state right after setting state, it will not be accurate. These checks should be done in a separate method, the method should return the correct className. Then call that method inside `render` to get the className and directly set it in the className attribute.